EduVPN-multiOS/Coordinators/AppCoordinator.swift

```diff
@@ -328,6 +329,46 @@ class AppCoordinator: RootViewCoordinator {
         }
     }
     
+    private func _scheduleCertificateExpirationNotification(for certificate: CertificateModel, on api: Api) {
+        guard let expirationDate = certificate.x509Certificate?.notAfter else { return }
+        guard let identifier = certificate.uniqueIdentifier else { return }
+        
+        let notificationTitle = NSLocalizedString("VPN certificate is expiring", comment: "")
+        
+        var notificationBody: String?
+        if let certificateTitle = api.instance?.displayNames?.localizedValue {
+            notificationBody = String.localizedStringWithFormat("Once expired the certificate for instance %@ needs to be refreshed.", certificateTitle)
+        }
+        
+        let notification = NotificationsService.makeNotification(title: notificationTitle, body: notificationBody)
+
+#if DEBUG
+        guard let expirationWarningDate = NSCalendar.current.date(byAdding: .second, value: 10, to: Date()) else { return }
+#else
+        guard let expirationWarningDate = NSCalendar.current.date(byAdding: .minute, value: -15, to: expirationDate), expirationDate.timeIntervalSince(Date()) < 0 else { return }
+#endif
+        let expirationWarningDateComponents = NSCalendar.current.dateComponents(in: NSTimeZone.default, from: expirationWarningDate)
+
+        os_log("Scheduling a cert expiration reminder for %{public}@ on %{public}@.", log: Log.general, type: .info, certificate.uniqueIdentifier ?? "", expirationDate.description)
+        NotificationsService.sendNotification(notification, withIdentifier: identifier, at: expirationWarningDateComponents) { error in
```

Comment:
May want to use dependency injection for this so it becomes easier to add tests in the future.